### DISCUSSION TITLE
Difference in waveform peak rendering between Wavesurfer 2.x and 7.x?

### DISCUSSION THREAD

**Discussion by b0ss-x** [katspaugh/wavesurfer.js#3333] [Q&A]
I'm migrating from wavesurfer 2.x to 7.x and I notice a big difference in how the waveforms are rendered. 

For example in 2.x:
![image](https://github.com/katspaugh/wavesurfer.js/assets/150304455/8554fca9-bb95-408b-a3d7-b82d562e0a04)

Looks like this in 7.x:
![image](https://github.com/katspaugh/wavesurfer.js/assets/150304455/fb47c304-ed72-4ea2-9089-6955a3d3ede0)

There are a lot less peaks rendered and I want it to look more like 2.x. My initialization code (using same widths/heights in both versions):
```
      const wavesurfer = WaveSurfer.create({
            container: waveform,
            waveColor: "white",
            progressColor: "#73f1e6",
            cursorColor: "transparent",
            barWidth: 3,
            barGap: 1,
            barRadius: 3,
            barHeight: 2,
            height: 50, 
            normalize: true,
            media: audio
        });
```

**Comment by katspaugh** [comment]
You should remove `barHeight: 2` as it stretches the waveform x2; and possibly also remove `normalize: true`.

**Reply by b0ss-x** [reply]
Thanks I tried removing barHeight and normalize, but the peaks are still not coming through like they did in 2.x
![image](https://github.com/katspaugh/wavesurfer.js/assets/150304455/e0e133ab-5ad8-4e34-a070-5c4479f982e4)

**Reply by katspaugh** [reply]
Can you upload and share your audio file?

**Reply by b0ss-x** [reply]
@katspaugh sure thing, please see attached
[sample.zip](https://github.com/katspaugh/wavesurfer.js/files/13343742/sample.zip)

**Reply by katspaugh** [reply]
Thanks! Yes, I can see it too. Looks like v2 used a different sampling algorithm (average values) than the current version (maximum values). I'll see if we can restore this behavior, perhaps as an option.

**Reply by b0ss-x** [reply]
That would be great appreciate it, we have a lot of audio in our site and the v2 sampling algorithm seems to reflect the changes in the audio better for us. Temporarily I'm working around it by using a custom `renderFunction` that is getting similar waveforms to v2:
```
renderFunction: (channels, ctx) => {
                const {
                    width,
                    height
                } = ctx.canvas;
                const channelData = channels[
                    0]; // Assuming mono for simplicity, or you could average the two channels
                const scale = channelData.length / width;
                const barWidth = 5; // Width of the bars as modified
                const gap = 1; // Gap between bars as modified
                const maxBarHeight = height * 3; // Maximum height of a bar

                // Function to draw bars
                const drawBars = (channelData, ctx, barWidth, gap, maxBarHeight) => {
                    for (let i = 0; i < width; i += barWidth + gap) {
                        const index = Math.floor(i * scale);
                        const normalizedValue = Math.abs(channelData[
                            index]); // Normalize value to be between 0 and 1
                        const barHeight = normalizedValue *
                            maxBarHeight; // Scale the value to the max bar height

                        // Draw the bar for this sample
                        ctx.fillRect(i, (height - barHeight) / 2, barWidth, barHeight);
                    }
                };

                // Clear the canvas
                ctx.clearRect(0, 0, width, height);

                // Set the style for the bars
                ctx.fillStyle = 'white';

                // Draw the bars for the waveform
                drawBars(channelData, ctx, barWidth, gap, maxBarHeight);
            },
```
